src/instruction/instructions/store/str.ts

```diff
@@ -25,19 +25,35 @@ export class StrImmediate5OffsetInstruction extends BaseInstruction {
   private rnPattern: string = '0110000000XXX000'
   private rtPattern: string = '0110000000000XXX'
   private immPattern: string = '01100XXXXX000000'
-  private expectedOptionCount: number = 3
+  private expectedOptionCountMin: number = 2
+  private expectedOptionCountMax: number = 3
+  private otherInstructionWithSameName: BaseInstruction[] = [
+    new StrRegisterOffsetInstruction()
+  ]
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     return (
       super.canEncodeInstruction(name, options) &&
-      isOptionCountValid(options, this.expectedOptionCount) &&
-      isImmediate(options[2]) &&
-      registerStringHasBrackets(options[1], options[2])
+      !this.otherInstructionWithSameName.some((instr) => {
+        return instr.canEncodeInstruction(name, options)
+      })
     )
   }
 
   public encodeInstruction(options: string[]): Halfword[] {
-    checkOptionCount(options, 3)
+    checkOptionCount(
+      options,
+      this.expectedOptionCountMin,
+      this.expectedOptionCountMax
+    )
+    checkBracketsOnLastOptions(
+      options,
+      this.expectedOptionCountMin,
+      this.expectedOptionCountMax
+    )
+    if (options.length == this.expectedOptionCountMin) {
```

Comment:
use `===` (strict equals operator) for comparison

Reply:
done also for load instructions and created issue for remaining code https://virtualctboard.atlassian.net/browse/VCB-215